streamer/src/nonblocking/quic.rs

```diff
@@ -126,83 +126,94 @@ async fn setup_connection(
-                    .unwrap(),
-                );
-                (connection_table_l, stake)
-            } else {
-                drop(staked_nodes);
-                let mut connection_table_l = connection_table.lock().unwrap();
-                if connection_table_l.total_size >= max_unstaked_connections {
-                    let max_connections = max_percentage_full.apply_to(max_unstaked_connections);
-                    let num_pruned = connection_table_l.prune_oldest(max_connections);
-                    stats.num_evictions.fetch_add(num_pruned, Ordering::Relaxed);
+            };
+
+            if stake != 0 || max_unstaked_connections > 0 {
+                if let Some((last_update, stream_exit)) = connection_table_l.try_add_connection(
+                    &remote_addr,
+                    Some(connection),
+                    timing::timestamp(),
+                    max_connections_per_ip,
+                ) {
+                    drop(connection_table_l);
+                    let stats = stats.clone();
+                    let connection_table1 = connection_table.clone();
+                    tokio::spawn(handle_connection(
+                        uni_streams,
+                        packet_sender,
+                        remote_addr,
+                        last_update,
+                        connection_table1,
+                        stream_exit,
+                        stats,
+                        stake,
+                    ));
+                } else {
+                    stats.connection_add_failed.fetch_add(1, Ordering::Relaxed);
                 }
-                connection.set_max_concurrent_uni_streams(
-                    VarInt::from_u64(QUIC_MAX_UNSTAKED_CONCURRENT_STREAMS as u64).unwrap(),
-                );
-                (connection_table_l, 0)
-            }
-        };
-
-        if stake != 0 || max_unstaked_connections > 0 {
-            if let Some((last_update, stream_exit)) = connection_table_l.try_add_connection(
-                &remote_addr,
-                Some(connection),
-                timing::timestamp(),
-                max_connections_per_ip,
-            ) {
-                drop(connection_table_l);
-                let stats = stats.clone();
-                let connection_table1 = connection_table.clone();
-                tokio::spawn(handle_connection(
-                    uni_streams,
-                    packet_sender,
-                    remote_addr,
-                    last_update,
-                    connection_table1,
-                    stream_exit,
-                    stats,
-                    stake,
-                ));
             } else {
-                stats.connection_add_failed.fetch_add(1, Ordering::Relaxed);
+                connection.close(0u32.into(), &[0u8]);
+                stats
+                    .connection_add_failed_unstaked_node
+                    .fetch_add(1, Ordering::Relaxed);
             }
         } else {
```

Comment:
this discards all errors.  for a follow up, this method should not be infallible

Reply:
Agreed, will do this in a followup.